multiple paket groups generate wrong dependencies at design time (inside VS)

### Description

If a project use multiple paket group for dependencies, the project build incorrectly in VS because load the wrong nuget dependencies.
it will build correctly from command line.

### Repro steps

A project (.net sdk) who have dependencies from multiple paket group inside the `paket.references`, and multiple target framework (let's say `net461` and `netstandard2.0`)

### Expected behavior

Each target framework (inside VS) will build with the correct dependencies of that target framework

### Actual behavior

All dependencies are added together, for all framework (not seperated by framework).
The restore target (inside .NET SdK) will just do a package resolution, so may use wrong packges version  and add not expected packages.
Some warning will be like:

```
warning NU1701: Package 'EventStore.Client 4.1.1-rc1' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.1'. This package may not be fully compatible with your project.
```
or

```
warning MSB3277: Found conflicts between different versions of "FSharp.Core" that could not be resolved.
```

### Affected version

from v5.174

### Known workarounds

Revert to paket version 5.173.4
Be sure to have aligned the `.paket\Paket.Restore.targets` and remove the `obj` dir

### additional info

The issue is that all dependencies are specified in `Marvel.Projection.fsproj.paket.props` (loaded only at design time) 

```xml
    <ItemGroup Condition="($(DesignTimeBuild) == true) AND (('$(TargetFramework)' == 'monoandroid') OR ('$(TargetFramework)' == 'monoandroid2.2') OR ('$(TargetFramework)' == 'monoandroid22') OR ('$(TargetFramework)' == 'monoandroid2.3') OR ('$(TargetFramework)' == 'monoandroid23') OR ('$(TargetFramework)' == 'monoandroid4.0.3') OR ('$(TargetFramework)' == 'monoandroid403') OR ('$(TargetFramework)' == 'monoandroid4.1') OR ('$(TargetFramework)' == 'monoandroid41') OR ('$(TargetFramework)' == 'monoandroid4.2') OR ('$(TargetFramework)' == 'monoandroid42') OR ('$(TargetFramework)' == 'monoandroid4.3') OR ('$(TargetFramework)' == 'monoandroid43') OR ('$(TargetFramework)' == 'monoandroid4.4') OR ('$(TargetFramework)' == 'monoandroid44') OR ('$(TargetFramework)' == 'monoandroid5.0') OR ('$(TargetFramework)' == 'monoandroid50') OR ('$(TargetFramework)' == 'monoandroid5.1') OR ('$(TargetFramework)' == 'monoandroid51') OR ('$(TargetFramework)' == 'monoandroid6.0') OR ('$(TargetFramework)' == 'monoandroid60') OR ('$(TargetFramework)' == 'monoandroid7.0') OR ('$(TargetFramework)' == 'monoandroid70') OR ('$(TargetFramework)' == 'monoandroid7.1') OR ('$(TargetFramework)' == 'monoandroid71') OR ('$(TargetFramework)' == 'monoandroid8.0') OR ('$(TargetFramework)' == 'monoandroid80') OR ('$(TargetFramework)' == 'monoandroid8.1') OR ('$(TargetFramework)' == 'monoandroid81') OR ('$(TargetFramework)' == 'monotouch') OR ('$(TargetFramework)' == 'native') OR ('$(TargetFramework)' == 'native(,arm)') OR ('$(TargetFramework)' == 'native(,Win32)') OR ('$(TargetFramework)' == 'native(,x64)') OR ('$(TargetFramework)' == 'native(Debug,arm)') OR ('$(TargetFramework)' == 'native(Debug,Win32)') OR ('$(TargetFramework)' == 'native(Debug,x64)') OR ('$(TargetFramework)' == 'native(Release,arm)') OR ('$(TargetFramework)' == 'native(Release,Win32)') OR ('$(TargetFramework)' == 'native(Release,x64)') OR ('$(TargetFramework)' == 'net10') OR ('$(TargetFramework)' == 'net11') OR ('$(TargetFramework)' == 'net20') OR ('$(TargetFramework)' == 'net30') OR ('$(TargetFramework)' == 'net35') OR ('$(TargetFramework)' == 'net35-Unity Full v3.5') OR ('$(TargetFramework)' == 'net35-Unity Full v35') OR ('$(TargetFramework)' == 'net35-Unity Micro v3.5') OR ('$(TargetFramework)' == 'net35-Unity Micro v35') OR ('$(TargetFramework)' == 'net35-Unity Subset v3.5') OR ('$(TargetFramework)' == 'net35-Unity Subset v35') OR ('$(TargetFramework)' == 'net35-Unity Web v3.5') OR ('$(TargetFramework)' == 'net35-Unity Web v35') OR ('$(TargetFramework)' == 'net40') OR ('$(TargetFramework)' == 'net403') OR ('$(TargetFramework)' == 'net45') OR ('$(TargetFramework)' == 'net451') OR ('$(TargetFramework)' == 'net452') OR ('$(TargetFramework)' == 'net453') OR ('$(TargetFramework)' == 'net46') OR ('$(TargetFramework)' == 'net461') OR ('$(TargetFramework)' == 'net462') OR ('$(TargetFramework)' == 'net463') OR ('$(TargetFramework)' == 'net47') OR ('$(TargetFramework)' == 'net471') OR ('$(TargetFramework)' == 'net472') OR ('$(TargetFramework)' == 'netcoreapp1.0') OR ('$(TargetFramework)' == 'netcoreapp10') OR ('$(TargetFramework)' == 'netcoreapp1.1') OR ('$(TargetFramework)' == 'netcoreapp11') OR ('$(TargetFramework)' == 'netcoreapp2.0') OR ('$(TargetFramework)' == 'netcoreapp20') OR ('$(TargetFramework)' == 'netcoreapp2.1') OR ('$(TargetFramework)' == 'netcoreapp21') OR ('$(TargetFramework)' == 'netstandard1.0') OR ('$(TargetFramework)' == 'netstandard10') OR ('$(TargetFramework)' == 'netstandard1.1') OR ('$(TargetFramework)' == 'netstandard11') OR ('$(TargetFramework)' == 'netstandard1.2') OR ('$(TargetFramework)' == 'netstandard12') OR ('$(TargetFramework)' == 'netstandard1.3') OR ('$(TargetFramework)' == 'netstandard13') OR ('$(TargetFramework)' == 'netstandard1.4') OR ('$(TargetFramework)' == 'netstandard14') OR ('$(TargetFramework)' == 'netstandard1.5') OR ('$(TargetFramework)' == 'netstandard15') OR ('$(TargetFramework)' == 'netstandard1.6') OR ('$(TargetFramework)' == 'netstandard16') OR ('$(TargetFramework)' == 'netstandard2.0') OR ('$(TargetFramework)' == 'netstandard20') OR ('$(TargetFramework)' == 'portable-net40+sl4') OR ('$(TargetFramework)' == 'portable-net40+sl4+win8+wp7') OR ('$(TargetFramework)' == 'portable-net40+sl4+win8+wp75') OR ('$(TargetFramework)' == 'portable-net40+sl4+win8+wp8') OR ('$(TargetFramework)' == 'portable-net40+sl5') OR ('$(TargetFramework)' == 'portable-net40+sl5+win8') OR ('$(TargetFramework)' == 'portable-net40+sl5+win8+wp8') OR ('$(TargetFramework)' == 'portable-net40+sl5+win8+wp8+wpa81') OR ('$(TargetFramework)' == 'portable-net40+sl5+win8+wpa81') OR ('$(TargetFramework)' == 'portable-net40+win8') OR ('$(TargetFramework)' == 'portable-net40+win8+wpa81') OR ('$(TargetFramework)' == 'portable-net403+sl4') OR ('$(TargetFramework)' == 'portable-net403+sl4+win8') OR ('$(TargetFramework)' == 'portable-net403+sl4+win8+wp7') OR ('$(TargetFramework)' == 'portable-net403+sl4+win8+wp75') OR ('$(TargetFramework)' == 'portable-net403+sl4+win8+wp8') OR ('$(TargetFramework)' == 'portable-net403+sl5') OR ('$(TargetFramework)' == 'portable-net403+sl5+win8') OR ('$(TargetFramework)' == 'portable-net403+sl5+win8+wp8') OR ('$(TargetFramework)' == 'portable-net403+sl5+win8+wp8+wpa81') OR ('$(TargetFramework)' == 'portable-net403+sl5+win8+wpa81') OR ('$(TargetFramework)' == 'portable-net403+win8') OR ('$(TargetFramework)' == 'portable-net403+win8+wpa81') OR ('$(TargetFramework)' == 'portable-net45+sl4') OR ('$(TargetFramework)' == 'portable-net45+sl4+win8') OR ('$(TargetFramework)' == 'portable-net45+sl4+win8+wp7') OR ('$(TargetFramework)' == 'portable-net45+sl4+win8+wp75') OR ('$(TargetFramework)' == 'portable-net45+sl4+win8+wp8') OR ('$(TargetFramework)' == 'portable-net45+sl5') OR ('$(TargetFramework)' == 'portable-net45+sl5+win8') OR ('$(TargetFramework)' == 'portable-net45+sl5+win8+wp8') OR ('$(TargetFramework)' == 'portable-net45+sl5+win8+wp8+wpa81') OR ('$(TargetFramework)' == 'portable-net45+sl5+win8+wpa81') OR ('$(TargetFramework)' == 'portable-net45+win8') OR ('$(TargetFramework)' == 'portable-net45+win8+wp8') OR ('$(TargetFramework)' == 'portable-net45+win8+wp8+wpa81') OR ('$(TargetFramework)' == 'portable-net45+win8+wpa81') OR ('$(TargetFramework)' == 'portable-net45+wp8') OR ('$(TargetFramework)' == 'portable-net451+win81') OR ('$(TargetFramework)' == 'portable-net451+win81+wpa81') OR ('$(TargetFramework)' == 'portable-win81+wp81') OR ('$(TargetFramework)' == 'portable-win81+wp81+wpa81') OR ('$(TargetFramework)' == 'portable-win81+wpa81') OR ('$(TargetFramework)' == 'portable-wp81+wpa81') OR ('$(TargetFramework)' == 'sl3') OR ('$(TargetFramework)' == 'sl4') OR ('$(TargetFramework)' == 'sl5') OR ('$(TargetFramework)' == 'uap10.0') OR ('$(TargetFramework)' == 'uap100') OR ('$(TargetFramework)' == 'uap10.0.15138') OR ('$(TargetFramework)' == 'uap10015138') OR ('$(TargetFramework)' == 'uap10.0.16299') OR ('$(TargetFramework)' == 'uap10016299') OR ('$(TargetFramework)' == 'uap10.0.16300') OR ('$(TargetFramework)' == 'uap10016300') OR ('$(TargetFramework)' == 'uap10.1') OR ('$(TargetFramework)' == 'uap101') OR ('$(TargetFramework)' == 'win10') OR ('$(TargetFramework)' == 'win8') OR ('$(TargetFramework)' == 'win81') OR ('$(TargetFramework)' == 'wp7') OR ('$(TargetFramework)' == 'wp71') OR ('$(TargetFramework)' == 'wp75') OR ('$(TargetFramework)' == 'wp8') OR ('$(TargetFramework)' == 'wp81') OR ('$(TargetFramework)' == 'wpa81') OR ('$(TargetFramework)' == 'xamarinios') OR ('$(TargetFramework)' == 'xamarinmac') OR ('$(TargetFramework)' == 'xamarintvos') OR ('$(TargetFramework)' == 'xamarinwatchos'))">
        <PackageReference Include="EventStore.Client">
            <Version>4.1.1-rc1</Version>
        </PackageReference>
        <PackageReference Include="FSharp.Core">
            <Version>3.1.2.5</Version>
        </PackageReference>
        <PackageReference Include="FSharp.Control.AsyncSeq">
            <Version>2.0.21</Version>
        </PackageReference>
        <PackageReference Include="FSharp.Core">
            <Version>4.3.4</Version>
        </PackageReference>
```

instead of 

```xml
    <ItemGroup Condition="($(DesignTimeBuild) == true) AND ('$(TargetFramework)' == 'net461')">
        <PackageReference Include="EventStore.Client">
            <Version>4.1.1-rc1</Version>
        </PackageReference>
        <PackageReference Include="FSharp.Core">
            <Version>3.1.2.5</Version>
        </PackageReference>
        <PackageReference Include="FSharp.Control.AsyncSeq">
            <Version>2.0.21</Version>
        </PackageReference>
    </ItemGroup>
    <ItemGroup Condition="($(DesignTimeBuild) == true) AND ('$(TargetFramework)' == 'netstandard2.0')">
        <PackageReference Include="FSharp.Core">
            <Version>4.3.4</Version>
        </PackageReference>
```

--- a/src/Paket.Core/Installation/RestoreProcess.fs
+++ b/src/Paket.Core/Installation/RestoreProcess.fs
@@ -272,8 +272,14 @@ let createPaketPropsFile (lockFile:LockFile) (cliTools:ResolvedPackage seq) (pac
         else
             packages
             |> Seq.map (fun ((groupName,packageName),_,_) -> 
-                let p = lockFile.Groups.[groupName].Resolution.[packageName]
-                let condition = Paket.Requirements.getExplicitRestriction p.Settings.FrameworkRestrictions      
+                let group = lockFile.Groups.[groupName]
+                let p = group.Resolution.[packageName]
+                let restrictions =
+                    match p.Settings.FrameworkRestrictions with
+                    | FrameworkRestrictions.ExplicitRestriction FrameworkRestriction.HasNoRestriction -> group.Options.Settings.FrameworkRestrictions
+                    | FrameworkRestrictions.ExplicitRestriction fw -> FrameworkRestrictions.ExplicitRestriction fw
+                    | _ -> group.Options.Settings.FrameworkRestrictions
+                let condition = restrictions |> getExplicitRestriction
                 p,condition)
             |> Seq.groupBy snd
             |> Seq.collect (fun (condition,packages) -> 

--- a/src/Paket.Core/Installation/RestoreProcess.fs
+++ b/src/Paket.Core/Installation/RestoreProcess.fs
@@ -283,7 +283,10 @@ let createPaketPropsFile (lockFile:LockFile) (cliTools:ResolvedPackage seq) (pac
                 p,condition)
             |> Seq.groupBy snd
             |> Seq.collect (fun (condition,packages) -> 
-                let condition = condition.ToMSBuildCondition()
+                let condition =
+                    match condition with
+                    | FrameworkRestriction.HasNoRestriction -> ""
+                    | restrictions -> restrictions.ToMSBuildCondition()
                 let condition =
                     if condition = "" || condition = "true" then "" else
                     sprintf " AND (%s)" condition

--- a/tests/Paket.Tests/InstallModel/PaketPropsTests.fs
+++ b/tests/Paket.Tests/InstallModel/PaketPropsTests.fs
@@ -9,6 +9,14 @@ open System.Xml.Linq
 let ns = "http://schemas.microsoft.com/developer/msbuild/2003"
 let xname name = XName.Get(name, ns)
 
+let checkTargetFrameworkRestriction r (itemGroup: XElement) =
+    match itemGroup.Attribute(XName.Get "Condition") with
+    | null -> Assert.Fail(sprintf "Expected attribute 'Condition' but element was '%A'" itemGroup)
+    | a ->
+        let condition = a.Value
+        let msbuildCond = r |> Paket.Requirements.getExplicitRestriction |> fun c -> c.ToMSBuildCondition()
+        StringAssert.Contains(msbuildCond, condition)
+
 let checkContainsPackageRefs pkgRefs (group: XElement) =
 
     let isPackageReference name (x: XElement) =
@@ -22,8 +30,8 @@ let checkContainsPackageRefs pkgRefs (group: XElement) =
     let hasVersion version (x: XElement) =
         x.Elements(xname "Version")
         |> Seq.tryHead
-        |> Option.map (fun x -> x.ToString() = version)
-        |> function Some true -> true | _ -> false
+        |> Option.map (fun x -> x.Value = version)
+        |> Option.exists id
 
     let packageRefs = group.Elements(xname "PackageReference") |> Seq.toList
     Assert.AreEqual(pkgRefs |> List.length, packageRefs |> Seq.length, (sprintf "%A" group))
@@ -36,7 +44,7 @@ let checkContainsPackageRefs pkgRefs (group: XElement) =
         match pkg with
         | Some p -> ()
         | None ->
-            Assert.Fail("expected package '%s' with version '%s' not found in '%A' group")
+            Assert.Fail(sprintf "expected package '%s' with version '%s' not found in '%A' group" pkgName pkgVersion group)
 
 
 [<Test>]
@@ -55,7 +63,8 @@ NUGET
   remote: https://api.nuget.org/v3/index.json
     FsCheck (2.8.2)
       FSharp.Core (>= 3.1.2.5)
-    FSharp.Core (4.3.4)"""
+    FSharp.Core (4.3.4)
+"""
 
     let refFileContent = """
 FSharp.Core
@@ -68,7 +77,7 @@ group Other1
 
     let lockFile = LockFile.Parse("", toLines lockFile)
 
-    let refFile = ReferencesFile.FromLines(toLines refFileContent) // .Groups.[Constants.MainDependencyGroup]
+    let refFile = ReferencesFile.FromLines(toLines refFileContent)
 
     let packages =
         [ for kv in refFile.Groups do
@@ -85,8 +94,69 @@ group Other1
     match itemGroups with
     | [groupMain; otherGroup] ->
         groupMain
-        |> checkContainsPackageRefs [ "FSharp.Core","1"; "Argu","2" ] 
+        |> checkTargetFrameworkRestriction (lockFile.Groups.[Constants.MainDependencyGroup].Options.Settings.FrameworkRestrictions)
+        groupMain
+        |> checkContainsPackageRefs [ "FSharp.Core","3.1.2.5"; "Argu","4.2.1" ] 
         otherGroup
-        |> checkContainsPackageRefs [ "FSharp.Core","1"; "Argu","2" ] 
+        |> checkTargetFrameworkRestriction (lockFile.Groups.[Domain.GroupName "Other1"].Options.Settings.FrameworkRestrictions)
+        otherGroup
+        |> checkContainsPackageRefs [ "FSharp.Core","4.3.4"; "FsCheck","2.8.2" ] 
+    | l ->
+        Assert.Fail(sprintf "expected two ItemGroup but was '%A'" l)
+
+[<Test>]
+let ``should create props file for design mode with single package restriction``() = 
+
+    let lockFile = """RESTRICTION: && (>= net461) (< net47)
+NUGET
+  remote: https://api.nuget.org/v3/index.json
+    Argu (4.2.1)
+      FSharp.Core (>= 3.1.2)
+    FSharp.Core (3.1.2.5)
+
+GROUP Other2
+RESTRICTION: >= netcoreapp2.0
+NUGET
+  remote: https://api.nuget.org/v3/index.json
+    FsCheck (2.8.2) - restriction: && (== netcoreapp2.1) (>= netcoreapp2.0)
+      FSharp.Core (>= 3.1.2.5)
+    FSharp.Core (4.3.4)
+"""
+
+    let refFileContent = """
+FSharp.Core
+Argu
+
+group Other2
+  FSharp.Core
+  FsCheck
+"""
+
+    let lockFile = LockFile.Parse("", toLines lockFile)
+
+    let refFile = ReferencesFile.FromLines(toLines refFileContent)
+
+    let packages =
+        [ for kv in refFile.Groups do
+            let packagesInGroup,_ = lockFile.GetOrderedPackageHull(kv.Key, refFile)
+            yield! packagesInGroup ]
+
+    let outPath = System.IO.Path.GetTempFileName()
+    Paket.RestoreProcess.createPaketPropsFile lockFile Seq.empty packages (FileInfo outPath)
+
+    let doc = XDocument.Load(outPath, LoadOptions.PreserveWhitespace)
+
+    let itemGroups = doc.Root.Elements (xname "ItemGroup") |> Seq.toList
+            
+    match itemGroups with
+    | [groupMain; otherGroup] ->
+        groupMain
+        |> checkTargetFrameworkRestriction (lockFile.Groups.[Constants.MainDependencyGroup].Options.Settings.FrameworkRestrictions)
+        groupMain
+        |> checkContainsPackageRefs [ "FSharp.Core","3.1.2.5"; "Argu","4.2.1" ] 
+        otherGroup
+        |> checkTargetFrameworkRestriction (lockFile.Groups.[Domain.GroupName "Other1"].Options.Settings.FrameworkRestrictions)
+        otherGroup
+        |> checkContainsPackageRefs [ "FSharp.Core","4.3.4"; "FsCheck","2.8.2" ] 
     | l ->
         Assert.Fail(sprintf "expected two ItemGroup but was '%A'" l)

--- a/tests/Paket.Tests/InstallModel/PaketPropsTests.fs
+++ b/tests/Paket.Tests/InstallModel/PaketPropsTests.fs
@@ -48,7 +48,7 @@ let checkContainsPackageRefs pkgRefs (group: XElement) =
 
 
 [<Test>]
-let ``should create props file for design mode``() = 
+let ``should create props file for design mode with group restrictions``() = 
 
     let lockFile = """RESTRICTION: && (>= net461) (< net47)
 NUGET

--- a/tests/Paket.Tests/InstallModel/PaketPropsTests.fs
+++ b/tests/Paket.Tests/InstallModel/PaketPropsTests.fs
@@ -221,3 +221,40 @@ group Other2
         |> checkContainsPackageRefs [ "FsCheck","2.8.2" ] 
     | l ->
         Assert.Fail(sprintf "expected three ItemGroup but was '%A'" l)
+
+[<Test>]
+let ``should create props file for design mode package restriction``() = 
+
+    let lockFile = """NUGET
+  remote: https://api.nuget.org/v3/index.json
+    Newtonsoft.Json (11.0.2) - restriction: == netcoreapp2.1
+"""
+
+    let refFileContent = """
+Newtonsoft.Json
+"""
+
+    let lockFile = LockFile.Parse("", toLines lockFile)
+
+    let refFile = ReferencesFile.FromLines(toLines refFileContent)
+
+    let packages =
+        [ for kv in refFile.Groups do
+            let packagesInGroup,_ = lockFile.GetOrderedPackageHull(kv.Key, refFile)
+            yield! packagesInGroup ]
+
+    let outPath = System.IO.Path.GetTempFileName()
+    Paket.RestoreProcess.createPaketPropsFile lockFile Seq.empty packages (FileInfo outPath)
+
+    let doc = XDocument.Load(outPath, LoadOptions.PreserveWhitespace)
+
+    let itemGroups = doc.Root.Elements (xname "ItemGroup") |> Seq.toList
+            
+    match itemGroups with
+    | [groupMain] ->
+        groupMain
+        |> checkTargetFrameworkRestriction (lockFile.Groups.[Constants.MainDependencyGroup].Resolution.[Domain.PackageName "Newtonsoft.Json"].Settings.FrameworkRestrictions)
+        groupMain
+        |> checkContainsPackageRefs [ "Newtonsoft.Json","11.0.2" ] 
+    | l ->
+        Assert.Fail(sprintf "expected one ItemGroup but was '%A'" l)

--- a/tests/Paket.Tests/InstallModel/PaketPropsTests.fs
+++ b/tests/Paket.Tests/InstallModel/PaketPropsTests.fs
@@ -149,14 +149,20 @@ group Other2
     let itemGroups = doc.Root.Elements (xname "ItemGroup") |> Seq.toList
             
     match itemGroups with
-    | [groupMain; otherGroup] ->
+    | [groupMain; otherGroup20And21; otherGroupOnly21] ->
         groupMain
         |> checkTargetFrameworkRestriction (lockFile.Groups.[Constants.MainDependencyGroup].Options.Settings.FrameworkRestrictions)
         groupMain
         |> checkContainsPackageRefs [ "FSharp.Core","3.1.2.5"; "Argu","4.2.1" ] 
-        otherGroup
-        |> checkTargetFrameworkRestriction (lockFile.Groups.[Domain.GroupName "Other1"].Options.Settings.FrameworkRestrictions)
-        otherGroup
-        |> checkContainsPackageRefs [ "FSharp.Core","4.3.4"; "FsCheck","2.8.2" ] 
+
+        otherGroup20And21
+        |> checkTargetFrameworkRestriction (lockFile.Groups.[Domain.GroupName "Other2"].Options.Settings.FrameworkRestrictions)
+        otherGroup20And21
+        |> checkContainsPackageRefs [ "FSharp.Core","4.3.4" ] 
+
+        otherGroupOnly21
+        |> checkTargetFrameworkRestriction (lockFile.Groups.[Domain.GroupName "Other2"].Resolution.[Domain.PackageName "FsCheck"].Settings.FrameworkRestrictions)
+        otherGroupOnly21
+        |> checkContainsPackageRefs [ "FsCheck","2.8.2" ] 
     | l ->
         Assert.Fail(sprintf "expected two ItemGroup but was '%A'" l)

--- a/tests/Paket.Tests/InstallModel/PaketPropsTests.fs
+++ b/tests/Paket.Tests/InstallModel/PaketPropsTests.fs
@@ -160,7 +160,7 @@ group Other1
         Assert.Fail(sprintf "expected two ItemGroup but was '%A'" l)
 
 [<Test>]
-let ``should create props file for design mode with single package restriction``() = 
+let ``should create props file for design mode with group and package restriction``() = 
 
     let lockFile = """RESTRICTION: && (>= net461) (< net47)
 NUGET

--- a/tests/Paket.Tests/InstallModel/PaketPropsTests.fs
+++ b/tests/Paket.Tests/InstallModel/PaketPropsTests.fs
@@ -1,0 +1,92 @@
+﻿module Paket.InstallModel.PaketPropsTests
+
+open Paket
+open NUnit.Framework
+open FsUnit
+open TestHelpers
+open System.Xml.Linq
+
+let ns = "http://schemas.microsoft.com/developer/msbuild/2003"
+let xname name = XName.Get(name, ns)
+
+let checkContainsPackageRefs pkgRefs (group: XElement) =
+
+    let isPackageReference name (x: XElement) =
+        if x.Name = (xname "PackageReference") then
+            match x.Attribute(XName.Get "Include") with
+            | null -> false
+            | v -> v.Value = name
+        else
+            false
+
+    let hasVersion version (x: XElement) =
+        x.Elements(xname "Version")
+        |> Seq.tryHead
+        |> Option.map (fun x -> x.ToString() = version)
+        |> function Some true -> true | _ -> false
+
+    let packageRefs = group.Elements(xname "PackageReference") |> Seq.toList
+    Assert.AreEqual(pkgRefs |> List.length, packageRefs |> Seq.length, (sprintf "%A" group))
+    for (pkgName, pkgVersion) in pkgRefs do
+        let pkg =
+            packageRefs
+            |> List.filter (isPackageReference pkgName)
+            |> List.filter (hasVersion pkgVersion)
+            |> List.tryHead
+        match pkg with
+        | Some p -> ()
+        | None ->
+            Assert.Fail("expected package '%s' with version '%s' not found in '%A' group")
+
+
+[<Test>]
+let ``should create props file for design mode``() = 
+
+    let lockFile = """RESTRICTION: && (>= net461) (< net47)
+NUGET
+  remote: https://api.nuget.org/v3/index.json
+    Argu (4.2.1)
+      FSharp.Core (>= 3.1.2)
+    FSharp.Core (3.1.2.5)
+
+GROUP Other1
+RESTRICTION: == netstandard2.0
+NUGET
+  remote: https://api.nuget.org/v3/index.json
+    FsCheck (2.8.2)
+      FSharp.Core (>= 3.1.2.5)
+    FSharp.Core (4.3.4)"""
+
+    let refFileContent = """
+FSharp.Core
+Argu
+
+group Other1
+  FSharp.Core
+  FsCheck
+"""
+
+    let lockFile = LockFile.Parse("", toLines lockFile)
+
+    let refFile = ReferencesFile.FromLines(toLines refFileContent) // .Groups.[Constants.MainDependencyGroup]
+
+    let packages =
+        [ for kv in refFile.Groups do
+            let packagesInGroup,_ = lockFile.GetOrderedPackageHull(kv.Key, refFile)
+            yield! packagesInGroup ]
+
+    let outPath = System.IO.Path.GetTempFileName()
+    Paket.RestoreProcess.createPaketPropsFile lockFile Seq.empty packages (FileInfo outPath)
+
+    let doc = XDocument.Load(outPath, LoadOptions.PreserveWhitespace)
+
+    let itemGroups = doc.Root.Elements (xname "ItemGroup") |> Seq.toList
+            
+    match itemGroups with
+    | [groupMain; otherGroup] ->
+        groupMain
+        |> checkContainsPackageRefs [ "FSharp.Core","1"; "Argu","2" ] 
+        otherGroup
+        |> checkContainsPackageRefs [ "FSharp.Core","1"; "Argu","2" ] 
+    | l ->
+        Assert.Fail(sprintf "expected two ItemGroup but was '%A'" l)

--- a/tests/Paket.Tests/InstallModel/PaketPropsTests.fs
+++ b/tests/Paket.Tests/InstallModel/PaketPropsTests.fs
@@ -9,13 +9,19 @@ open System.Xml.Linq
 let ns = "http://schemas.microsoft.com/developer/msbuild/2003"
 let xname name = XName.Get(name, ns)
 
-let checkTargetFrameworkRestriction r (itemGroup: XElement) =
+let checkTargetFrameworkCondition msbuildCondition (itemGroup: XElement) =
     match itemGroup.Attribute(XName.Get "Condition") with
     | null -> Assert.Fail(sprintf "Expected attribute 'Condition' but element was '%A'" itemGroup)
     | a ->
         let condition = a.Value
-        let msbuildCond = r |> Paket.Requirements.getExplicitRestriction |> fun c -> c.ToMSBuildCondition()
-        StringAssert.Contains(msbuildCond, condition)
+        Assert.AreEqual("($(DesignTimeBuild) == true)" + msbuildCondition, condition)
+
+let checkTargetFrameworkNoRestriction itemGroup =
+    checkTargetFrameworkCondition "" itemGroup
+
+let checkTargetFrameworkRestriction r itemGroup =
+    let msbuildCond = r |> Paket.Requirements.getExplicitRestriction |> fun c -> c.ToMSBuildCondition()
+    checkTargetFrameworkCondition (sprintf " AND (%s)" msbuildCond) itemGroup
 
 let checkContainsPackageRefs pkgRefs (group: XElement) =
 
@@ -46,6 +52,55 @@ let checkContainsPackageRefs pkgRefs (group: XElement) =
         | None ->
             Assert.Fail(sprintf "expected package '%s' with version '%s' not found in '%A' group" pkgName pkgVersion group)
 
+
+[<Test>]
+let ``should create props file for design mode``() = 
+
+    let lockFile = """NUGET
+  remote: https://api.nuget.org/v3/index.json
+    Argu (4.2.1)
+      FSharp.Core (>= 3.1.2)
+    FSharp.Core (3.1.2.5)
+
+GROUP Other1
+NUGET
+  remote: https://api.nuget.org/v3/index.json
+    FsCheck (2.8.2)
+      FSharp.Core (>= 3.1.2.5)
+"""
+
+    let refFileContent = """
+FSharp.Core
+Argu
+
+group Other1
+  FsCheck
+"""
+
+    let lockFile = LockFile.Parse("", toLines lockFile)
+
+    let refFile = ReferencesFile.FromLines(toLines refFileContent)
+
+    let packages =
+        [ for kv in refFile.Groups do
+            let packagesInGroup,_ = lockFile.GetOrderedPackageHull(kv.Key, refFile)
+            yield! packagesInGroup ]
+
+    let outPath = System.IO.Path.GetTempFileName()
+    Paket.RestoreProcess.createPaketPropsFile lockFile Seq.empty packages (FileInfo outPath)
+
+    let doc = XDocument.Load(outPath, LoadOptions.PreserveWhitespace)
+
+    let itemGroups = doc.Root.Elements (xname "ItemGroup") |> Seq.toList
+            
+    match itemGroups with
+    | [groupMain] ->
+        groupMain
+        |> checkTargetFrameworkNoRestriction
+        groupMain
+        |> checkContainsPackageRefs [ "FSharp.Core","3.1.2.5"; "Argu","4.2.1"; "FsCheck","2.8.2" ] 
+    | l ->
+        Assert.Fail(sprintf "expected one ItemGroup but was '%A'" l)
 
 [<Test>]
 let ``should create props file for design mode with group restrictions``() = 
@@ -165,4 +220,4 @@ group Other2
         otherGroupOnly21
         |> checkContainsPackageRefs [ "FsCheck","2.8.2" ] 
     | l ->
-        Assert.Fail(sprintf "expected two ItemGroup but was '%A'" l)
+        Assert.Fail(sprintf "expected three ItemGroup but was '%A'" l)

--- a/tests/Paket.Tests/Paket.Tests.fsproj
+++ b/tests/Paket.Tests/Paket.Tests.fsproj
@@ -353,6 +353,7 @@
     <Compile Include="InstallModel\BindingRedirect.fs" />
     <Compile Include="InstallModel\RuntimeGraphTests.fs" />
     <Compile Include="InstallModel\InstallNuGetContentsTests.fs" />
+    <Compile Include="InstallModel\PaketPropsTests.fs" />
     <Compile Include="Packaging\PackageProcessSpecs.fs" />
     <Compile Include="Packaging\NuspecWriterSpecs.fs" />
     <Compile Include="Packaging\TemplateFileParsing.fs" />

